src/Node/Stream.js

```diff
@@ -12,26 +12,30 @@ exports.setEncodingImpl = function(s) {
     };
 };
 
-exports.onDataEitherImpl = function(left){
-    return function(right){
-        return function(s) {
-            return function(f) {
-                return function() {
-                    s.on('data', function(chunk) {
-                        if (chunk instanceof Buffer) {
-                            f(right(chunk))();
-                        }
-                        else if (typeof chunk === "string") {
-                            f(left(chunk))();
-                        }
-                        else {
-                            throw new Error(
-                                "Node.Stream.onDataEitherImpl: Unrecognised" +
-                                "chunk type; expected String or Buffer, got:" +
-                                chunk);
-                        }
-                    });
-                };
+exports.readChunkImpl = function(Left) {
+    return function(Right) {
+        return function(chunk) {
+            if (chunk instanceof Buffer) {
+                return Right(chunk);
+            } else if (typeof chunk === 'string') {
+                return Left(chunk);
+            } else {
+                throw new Error(
+                  "Node.Stream.readChunkImpl: Unrecognised " +
+                  "chunk type; expected String or Buffer, got: " +
+                  chunk);
+            }
+        };
+    };
+};
+
+exports.onDataImpl = function(readChunk) {
+    return function(r) {
+        return function(f) {
+            return function() {
+                r.on('data', function(data) {
+                    f(readChunk(data));
```

Comment:
Hmmm, actually not sure how to test this, since we are not using `Aff` or `ContT`. In other words, if no callback is called, `Eff` don't care.